### ISSUE TITLE
ISearch Backward Text conflict.

### ISSUE THREAD

**Issue by charjr** [strindberg/emacsj#41]
_ISearch Backward Text_ appears to be conflicting with _Search In Command History_ in a terminal. 
Removing/changing the binding for _Search In Command History_ makes it work as expected.
This has only been an issue for me as of version 2024.1

![image](https://github.com/strindberg/emacsj/assets/102669158/f8faa839-597e-4c7f-846b-ddc65786c2e5)

Other than that, brilliant plug in! Thank you for your hard work.

**Comment by strindberg** [comment]
Thank you very much for your report! I see it as well, but it must be some new addition in 2024.1. I will investigate.

**Comment by strindberg** [comment]
After further investigation, I am doubtful I should do anything about this in the EmacsJ plugin. The ctrl-r binding seems to be a new addition to the Terminal plugin in 2024.1.

This is a genuine clash of traditions, in my view: ctrl-r is bound to isearch-backward in Emacs, but it is also the standard keybinding for history search in bash, zsh and other shells. Starting a terminal within an editor puts these two traditions/contexts into conflict.

My mantra in developing this plugin has often been "when in doubt, do as Emacs does it", and I believe Emacs actually uses ctrl-r for history search if one starts a terminal within Emacs: using it for isearch-backward requires a rebinding of the key.

Furthermore, it would feel a bit presumptuous for me to override the key bindings of another plugin: it seems more reasonable that users of both plugins will resolve such conflicts manually. I actually don't know exactly what it is that makes the terminal plugin's key binding take precedence over EmacsJ's, and truth be told I haven't even figured out exactly how this keybinding is registered by the terminal plugin; it seems to use some mechanism that I haven't seen before.

But in any case, for the above reasons, I believe that we should leave the plugin as it is, and that people who, like you, want to use ctrl-r for isearch-backward within the terminal can reconfigure the keys.

That said, I'd be open for further discussion, of course.

**Comment by charjr** [comment]
I agree, `ctrl+r` to search history in terminal makes sense. But let me elaborate further on the conflict I was experiencing.

The clash I was having is: If I pressed `ctrl+r` **outside the terminal** it pops up the isearch-backward prompt, this is what I expect.

Still **outside a terminal**, once the isearch-backward prompt is displayed, pressing `ctrl+r` a second time closes it, this is not what I expect. 

This means I cannot successively perform isearch-backwards, I can only perform the action once.

My short-term fix has been to remove the `ctrl+r` keybinding for terminal history searching which makes isearch-backward work as expected.

I also do not know why a terminal keybinding is conflicting with bindings outside of the terminal (And I'm afraid I have no experience writing IntelliJ plugins so I do not know how to fix it either)

Either way, thank you for the fantastic plugin and I'll be continuing to use it!

**Comment by strindberg** [comment]
I see. Thank you for the clarification. This is worse than I thought, of course. At the same time, I cannot right away see why the Teminal plugin would "steal" a key binding in this way and if it's the fault of my plugin or not. I'll have to reflect on this.

**Comment by charjr** [comment]
@strindberg Not helpful as a fix, but maybe helpful to anyone with the same issue:

Strangely enough if I **remove the terminal keybinding, it still works** but it also stops stealing the keybinding from EmacsJ.

Maybe it's a specific combination of settings I've got, but I don't know.

**Comment by strindberg** [comment]
To add not-helpful notes to this issue: I just noticed that `ctrl+r`behaves the same as `ctrl+t`: the Terminal plugin steals the shortcut if it's bound to a terminal shortcut (delete word backwards by default). De-activating the terminal's shortcut also restores the functionality. I will file a bug report with the Terminal plugin.

**Comment by strindberg** [comment]
I close this issue. There is something wrong with the way the Terminal plugin steals key strokes from other plugins, but I don't know enough about that plugin to figure out why, and the Youtrack issue hasn't seen much action. In the new release (1.3.3) I have inactivated the two problematic Terminal key bindings in the EmacsJ keymap.
